ui/app/AppLayouts/Wallet/popups/SignTransactionModalBase.qml

```diff
@@ -0,0 +1,174 @@
+                objectName: "signButton"
+                id: signButton
+                interactive: !root.loading
+                icon.name: Constants.authenticationIconByType[root.loginType]
+                text: qsTr("Sign")
+                onClicked: root.accept() // close and emit accepted() signal
+            }
+        }
+    }
+
+    property color gradientColor: backgroundColor
+    property url fromImageSource
+    property url toImageSource
+    property alias headerMainText: headerMainText.text
+    property alias headerSubTextLayout: headerSubTextLayout.children
+    property string infoTagText
+
+    default property alias contents: contentsLayout.children
+
+    width: 480
+    padding: 0
+
+    function formatBigNumber(number: string, decimals = -1) {
+        if (!number)
+            return ""
+        const big = SQUtils.AmountsArithmetic.fromString(number)
+        const resultNum = decimals === -1 ? big.toFixed() : big.round(decimals).toFixed()
+        return resultNum.replace('.', Qt.locale().decimalPoint)
+    }
+
+    header: StatusDialogHeader {
+        visible: root.title || root.subtitle
+        headline.title: root.title
+        headline.subtitle: root.subtitle
+        actions.closeButton.onClicked: root.closeHandler()
+
+        leftComponent: root.headerIconComponent
+    }
+
+    footer: StatusDialogFooter {
+        dropShadowEnabled: true
+
+        leftButtons: root.leftFooterContents
+        rightButtons: root.rightFooterContents
+    }
+
+    StatusScrollView {
+        id: scrollView
+        anchors.fill: parent
+        contentWidth: availableWidth
+        topPadding: 0
+        bottomPadding: 0
+
+        ColumnLayout {
+            anchors.left: parent.left
+            anchors.leftMargin: 4
+            anchors.right: parent.right
+            anchors.rightMargin: 4
+            spacing: 0
+
+            // header box with gradient
+            Rectangle {
+                Layout.fillWidth: true
+                Layout.leftMargin: -parent.anchors.leftMargin - scrollView.leftPadding
+                Layout.rightMargin: -parent.anchors.rightMargin - scrollView.rightPadding
+                Layout.preferredHeight: 266 // design
+                gradient: Gradient {
+                    GradientStop { position: 0.0; color: root.gradientColor }
+                    GradientStop { position: 1.0; color: root.backgroundColor }
+                }
+
+                ColumnLayout {
+                    width: 336 // by design
+                    spacing: 12
+                    anchors.centerIn: parent
+
+                    Row {
+                        Layout.alignment: Qt.AlignHCenter
+                        Layout.topMargin: 4
+                        spacing: -10
+                        StatusRoundedImage {
+                            objectName: "fromImage"
+                            width: 42
+                            height: 42
+                            border.width: 2
```

Comment:
It reminded me https://github.com/status-im/status-desktop/pull/15127/files#diff-499740ef32d0aa5a91396a3ae11f762064b70981fe659c3087f6bfd6a2dc789fR140 Here the solution is very simple and elegant, I was thinking if it could be used also in the old pr - probably yes but not easily because background color is not under full control like in the case here.